## Exploración de datos

### Índice
1. [Descripción de los datos ](#id1)
2. [Outliers](#id2)
3. [Distribución de los datos](#id3)
4. [Visualización de los datos](#id4)
5. [Conclusión](#id5)

### 1. Descripción de los datos<a name="id1"></a>
Se describen la cantidad de registros y columnas con sus tipos de datos, ocupación en memoria y una estadística descriptiva de los mismos: cúal es el promedio, el desvío estandar, los cuartiles, todos medidas de tendencia central.

In [3]:
# se importan las librerias necesarias
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np


In [4]:
# Se importa el dataset de trabajo
df = pd.read_csv('./dataset/movies_limpio.csv',  delimiter=',')

In [10]:
df.head() # Se observan los primeros registros

belongs_to_collection      budget  \
0  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000.0   
1                                                NaN  65000000.0   
2  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0.0   
3                                                NaN  16000000.0   
4  {'id': 96871, 'name': 'Father of the Bride Col...         0.0   

                                              genres     id original_language  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...    862                en   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   8844                en   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...  15602                en   
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...  31357                en   
4                     [{'id': 35, 'name': 'Comedy'}]  11862                en   

                                            overview  popularity  \
0  Led by Woody, Andy's toys live happily in his ...   21.946943   
1  When siblings Judy and Peter discover an encha...   17.015539   
2  A family wedding reignites the ancient feud be...   11.712900   
3  Cheated on, mistreated and stepped on, the wom...    3.859495   
4  Just when George Banks has recovered from his ...    8.387519   

                                production_companies  \
0     [{'name': 'Pixar Animation Studios', 'id': 3}]   
1  [{'name': 'TriStar Pictures', 'id': 559}, {'na...   
2  [{'name': 'Warner Bros.', 'id': 6194}, {'name'...   
3  [{'name': 'Twentieth Century Fox Film Corporat...   
4  [{'name': 'Sandollar Productions', 'id': 5842}...   

                                production_countries release_date  ...  \
0  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-10-30  ...   
1  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-15  ...   
2  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22  ...   
3  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22  ...   
4  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-02-10  ...   

     status                                            tagline  \
0  Released                                                NaN   
1  Released          Roll the dice and unleash the excitement!   
2  Released  Still Yelling. Still Fighting. Still Ready for...   
3  Released  Friends are the people who let you be yourself...   
4  Released  Just When His World Is Back To Normal... He's ...   

                         title vote_average vote_count  \
0                    Toy Story          7.7     5415.0   
1                      Jumanji          6.9     2413.0   
2             Grumpier Old Men          6.5       92.0   
3            Waiting to Exhale          6.1       34.0   
4  Father of the Bride Part II          5.7      173.0   

                       franquicia  \
0            Toy Story Collection   
1                             NaN   
2       Grumpy Old Men Collection   
3                             NaN   
4  Father of the Bride Collection   

                                         productoras  \
0                            Pixar Animation Studios   
1  TriStar Pictures, Teitler Film, Interscope Com...   
2                       Warner Bros., Lancaster Gate   
3             Twentieth Century Fox Film Corporation   
4         Sandollar Productions, Touchstone Pictures   

                     paises release_year    retorno  
0  United States of America         1995  12.451801  
1  United States of America         1995   4.043035  
2  United States of America         1995   0.000000  
3  United States of America         1995   5.090760  
4  United States of America         1995   0.000000  

[5 rows x 23 columns]

In [21]:
df.columns # se listan las columnas

Index(['belongs_to_collection', 'budget', 'genres', 'id', 'original_language',
       'overview', 'popularity', 'production_companies',
       'production_countries', 'release_date', 'revenue', 'runtime',
       'spoken_languages', 'status', 'tagline', 'title', 'vote_average',
       'vote_count', 'franquicia', 'productoras', 'paises', 'release_year',
       'retorno'],
      dtype='object')

In [22]:
print('Cantidad de Filas y Columnas',df.shape)

Cantidad de Filas y Columnas (45376, 23)


In [17]:
df.describe(include="all")  # Estadistica descriptiva
# Esto es un resumen del conjunto de datos para todas las columnas numéricas, excepto los valores NaN

belongs_to_collection        budget  \
count                                                4488  4.537600e+04   
unique                                               1695           NaN   
top     {'id': 415931, 'name': 'The Bowery Boys', 'pos...           NaN   
freq                                                   29           NaN   
mean                                                  NaN  4.232604e+06   
std                                                   NaN  1.743986e+07   
min                                                   NaN  0.000000e+00   
25%                                                   NaN  0.000000e+00   
50%                                                   NaN  0.000000e+00   
75%                                                   NaN  0.000000e+00   
max                                                   NaN  3.800000e+08   

                               genres             id original_language  \
count                           45376   45376.000000             45365   
unique                           4065            NaN                89   
top     [{'id': 18, 'name': 'Drama'}]            NaN                en   
freq                             4998            NaN             32202   
mean                              NaN  108027.103094               NaN   
std                               NaN  112168.377996               NaN   
min                               NaN       2.000000               NaN   
25%                               NaN   26385.750000               NaN   
50%                               NaN   59857.500000               NaN   
75%                               NaN  156533.500000               NaN   
max                               NaN  469172.000000               NaN   

                  overview    popularity production_companies  \
count                44435  45376.000000                45376   
unique               44232           NaN                22703   
top     No overview found.           NaN                   []   
freq                   133           NaN                11796   
mean                   NaN      2.926458                  NaN   
std                    NaN      6.009672                  NaN   
min                    NaN      0.000000                  NaN   
25%                    NaN      0.388840                  NaN   
50%                    NaN      1.130454                  NaN   
75%                    NaN      3.691695                  NaN   
max                    NaN    547.488298                  NaN   

                                     production_countries release_date  ...  \
count                                               45376        45376  ...   
unique                                               2389        17333  ...   
top     [{'iso_3166_1': 'US', 'name': 'United States o...   2008-01-01  ...   
freq                                                17846          136  ...   
mean                                                  NaN          NaN  ...   
std                                                   NaN          NaN  ...   
min                                                   NaN          NaN  ...   
25%                                                   NaN          NaN  ...   
50%                                                   NaN          NaN  ...   
75%                                                   NaN          NaN  ...   
max                                                   NaN          NaN  ...   

          status                 tagline       title  vote_average  \
count      45296                   20398       45376  45376.000000   
unique         6                   20269       42196           NaN   
top     Released  Based on a true story.  Cinderella           NaN   
freq       44936                       7          11           NaN   
mean         NaN                     NaN         NaN      5.624070   
std          NaN                     NaN         NaN      1.915423   
min          NaN          

In [20]:
df.describe() # describe reducido a las columnas numericas

budget             id    popularity       revenue       runtime  \
count  4.537600e+04   45376.000000  45376.000000  4.537600e+04  45130.000000   
mean   4.232604e+06  108027.103094      2.926458  1.123010e+07     94.181675   
std    1.743986e+07  112168.377996      6.009672  6.438996e+07     38.341059   
min    0.000000e+00       2.000000      0.000000  0.000000e+00      0.000000   
25%    0.000000e+00   26385.750000      0.388840  0.000000e+00     85.000000   
50%    0.000000e+00   59857.500000      1.130454  0.000000e+00     95.000000   
75%    0.000000e+00  156533.500000      3.691695  0.000000e+00    107.000000   
max    3.800000e+08  469172.000000    547.488298  2.787965e+09   1256.000000   

       vote_average    vote_count  release_year       retorno  
count  45376.000000  45376.000000  45376.000000  4.537600e+04  
mean       5.624070    110.096439   1991.881193  6.600428e+02  
std        1.915423    491.742894     24.055360  7.469329e+04  
min        0.000000      0.000000   1874.000000  0.000000e+00  
25%        5.000000      3.000000   1978.000000  0.000000e+00  
50%        6.000000     10.000000   2001.000000  0.000000e+00  
75%        6.800000     34.000000   2010.000000  0.000000e+00  
max       10.000000  14075.000000   2020.000000  1.239638e+07

Con los análisis anteriores se puede observar que las peliculas registras datan de los años 1874 al 2020. Entre ellas hay filmaciones con presupuesto y ganancia en cero. 
También se observa que hay peliculas que no tienen valoración registrada, en cuanto a puntuación por polularidad TMDB (TheMoviesDataBase), reseña y votación.

Se muestra cúales son esos registros:

In [23]:
# Se observa por columna: la cantidad de registro, el tipo de dato y si tiene valores nulos o no. 
# También se observa el uso de memoria.
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45376 entries, 0 to 45375
Data columns (total 23 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   belongs_to_collection  4488 non-null   object 
 1   budget                 45376 non-null  float64
 2   genres                 45376 non-null  object 
 3   id                     45376 non-null  int64  
 4   original_language      45365 non-null  object 
 5   overview               44435 non-null  object 
 6   popularity             45376 non-null  float64
 7   production_companies   45376 non-null  object 
 8   production_countries   45376 non-null  object 
 9   release_date           45376 non-null  object 
 10  revenue                45376 non-null  float64
 11  runtime                45130 non-null  float64
 12  spoken_languages       45376 non-null  object 
 13  status                 45296 non-null  object 
 14  tagline                20398 non-null  object 
 15  ti

In [18]:
df.isnull().sum() # se enlista los nulos por columna, pero no se los imputa.

belongs_to_collection    40888
budget                       0
genres                       0
id                           0
original_language           11
overview                   941
popularity                   0
production_companies         0
production_countries         0
release_date                 0
revenue                      0
runtime                    246
spoken_languages             0
status                      80
tagline                  24978
title                        0
vote_average                 0
vote_count                   0
franquicia               40888
productoras              12280
paises                    6216
release_year                 0
retorno                      0
dtype: int64

### 2. Ortliers<a name="id2"></a>
Detección de valores atípicos.